src/backend/core/api/viewsets.py

```diff
@@ -165,6 +184,12 @@ class RoomViewSet(
     permission_classes = [permissions.RoomPermissions]
     queryset = models.Room.objects.all()
     serializer_class = serializers.RoomSerializer
+    session_manager = RecordingSessionManager(
+        worker_class=LivekitEgressWorker,
```

Comment:
You are loosing the generic side of the `RecordingSessionManager` here.

Two solutions:
- use a setting along with `django.utils.import_string`
- use a setting and allow `RecordingSessionManager` to have a list of factories to instantiate the class itself.

I would use the first one, possibly with:

``` python
class RecordingSessionManager:
    def __init__(self):
        # settings.RECORDING_WORKER_CLASS = "core.recording.worker.LivekitEgressWorker"
        self._worker: type[AbstractRecordingWorker] = import_string(settings.RECORDING_WORKER_CLASS)()
```

Reply:
I have enhanced this part and refactored it.